src/cpp/fastdds/domain/DomainParticipantImpl.cpp

```diff
@@ -583,6 +583,21 @@ ContentFilteredTopic* DomainParticipantImpl::create_contentfilteredtopic(
         return nullptr;
     }
 
+    if (expression_parameters.size() > qos_.allocation().content_filter.expression_parameters.maximum)
+    {
+        EPROSIMA_LOG_ERROR(PARTICIPANT, "Number of expression parameters exceeds maximum allocation limit: "
+                << expression_parameters.size() << " > "
+                << qos_.allocation().content_filter.expression_parameters.maximum);
+        return nullptr;
+    }
+
+    if (expression_parameters.size() > 100)
+    {
+        EPROSIMA_LOG_ERROR(PARTICIPANT, "Number of expression parameters exceeds maximum protocol limit: "
+                << expression_parameters.size() << " > 100");
+        return nullptr;
+    }
```

Comment:
This limitation is not documented. We should add a documentation PR stating this limitation. I have added the corresponding label.